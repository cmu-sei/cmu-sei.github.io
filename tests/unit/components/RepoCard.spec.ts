import { shallowMount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import Component from '../../../components/RepoCard.vue'

describe('RepoCard.vue', () => {
  it('matches snapshot with default props', () => {
    const wrapper = shallowMount(Component)
    expect(wrapper.html()).toMatchSnapshot('defaultProps')
  })

  it('matches snapshot with set props', () => {
    const props = {
      item: {
        owner: '<OWNER>',
        ownerUrl: 'https://github.com/<OWNER>',
        name: '<NAME>',
        url: 'https://github.com/<OWNER>/<NAME>',
        description: 'Nulla quis lorem ut libero malesuada feugiat.',
        stars: 150,
        forks: 25,
        language: {
          title: 'Python',
          color: '#3572A5'
        }
      }
    }
    const wrapper = shallowMount(Component, { props })
    expect(wrapper.html()).toMatchSnapshot('withProps')
  })
})
