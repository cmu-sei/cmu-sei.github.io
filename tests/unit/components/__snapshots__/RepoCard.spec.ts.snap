// Vitest Snapshot v1

exports[`RepoCard.vue > matches snapshot with default props > defaultProps 1`] = `
"<div class=\\"border bg-white p-4 h-48 flex flex-col\\">
  <h4 class=\\"flex gap-2\\">
    <nuxt-link-stub class=\\"link link-danger font-bold\\"></nuxt-link-stub><span>/</span>
    <nuxt-link-stub class=\\"link link-danger\\"></nuxt-link-stub>
  </h4>
  <p class=\\"text-sm grow\\"></p>
  <div class=\\"flex gap-4 font-bold text-sm\\">
    <div class=\\"flex gap-1\\"><svg class=\\"w-4 h-4 mt-0.5\\">
        <use xlink:href=\\"#star\\"></use>
      </svg><span>0</span></div>
    <div class=\\"flex gap-1\\"><svg class=\\"w-4 h-4 mt-0.5\\">
        <use xlink:href=\\"#fork\\"></use>
      </svg><span>0</span></div>
    <div class=\\"flex gap-1\\"><i style=\\"background-color: #fff !important;\\" class=\\"block w-4 h-4 rounded-full bg-black mt-0.5\\" aria-hidden=\\"true\\"></i><span></span></div>
  </div>
</div>"
`;

exports[`RepoCard.vue > matches snapshot with set props > withProps 1`] = `
"<div class=\\"border bg-white p-4 h-48 flex flex-col\\">
  <h4 class=\\"flex gap-2\\">
    <nuxt-link-stub href=\\"https://github.com/&lt;OWNER&gt;\\" class=\\"link link-danger font-bold\\"></nuxt-link-stub><span>/</span>
    <nuxt-link-stub href=\\"https://github.com/&lt;OWNER&gt;/&lt;NAME&gt;\\" class=\\"link link-danger\\"></nuxt-link-stub>
  </h4>
  <p class=\\"text-sm grow\\">Nulla quis lorem ut libero malesuada feugiat.</p>
  <div class=\\"flex gap-4 font-bold text-sm\\">
    <div class=\\"flex gap-1\\"><svg class=\\"w-4 h-4 mt-0.5\\">
        <use xlink:href=\\"#star\\"></use>
      </svg><span>150</span></div>
    <div class=\\"flex gap-1\\"><svg class=\\"w-4 h-4 mt-0.5\\">
        <use xlink:href=\\"#fork\\"></use>
      </svg><span>25</span></div>
    <div class=\\"flex gap-1\\"><i style=\\"background-color: #3572A5 !important;\\" class=\\"block w-4 h-4 rounded-full bg-black mt-0.5\\" aria-hidden=\\"true\\"></i><span>Python</span></div>
  </div>
</div>"
`;
